packages/web-components/fast-components/src/disclosure/index.ts

```diff
@@ -0,0 +1,97 @@
+import { attr, customElement } from "@microsoft/fast-element";
+import { Disclosure, DisclosureTemplate as template } from "@microsoft/fast-foundation";
+import { DisclosureStyles as styles } from "./disclosure.styles";
+/**
+ * Types of anchor appearance.
+ * @public
+ */
+export type DisclosureAppearance = "accent" | "hypertext";
+
+/**
+ * The FAST Disclosure Element. Implements {@link @microsoft/fast-foundation#Disclosure},
+ * {@link @microsoft/fast-foundation#DisclosureTemplate}
+ *
+ *
+ * @public
+ * @remarks
+ * HTML Element: \<fast-Disclosure\>
+ *
+ */
+@customElement({
+    name: "fast-disclosure",
+    template,
+    styles,
+})
+export class FASTDisclosure extends Disclosure {
+    /**
+     * Disclosure default height
+     */
+    private height: number;
+    /**
+     * Disclosure height after it's expanded
+     */
+    private totalHeight: number;
+
+    /**
+     * The appearance the anchor should have.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: appearance
+     */
+    @attr
+    public appearance: DisclosureAppearance;
+    public appearanceChanged(
+        oldValue: DisclosureAppearance,
+        newValue: DisclosureAppearance
+    ): void {
+        if (oldValue !== newValue) {
+            this.classList.add(newValue);
+            this.classList.remove(oldValue);
+        }
+    }
+
+    /**
+     * Set disclosure height while transitioning
+     * @override
+     */
+    onToggle() {
+        super.onToggle();
+        this.details.style.setProperty("height", `${this.disclosureHeight}px`);
+    }
+
+    /**
+     * Calculate disclosure height before and after expanded
+     * @override
+     */
+    setup() {
+        super.setup();
+        if (!this.appearance) {
+            this.appearance = "accent";
+        }
+
+        const getCurrentHeight = () => this.details.getBoundingClientRect().height;
+        if (this.expanded) {
+            this.totalHeight = getCurrentHeight();
+            this.hide();
+            this.height = getCurrentHeight();
+            this.show();
+        } else {
+            this.height = getCurrentHeight();
+            this.show();
+            this.totalHeight = getCurrentHeight();
+            this.hide();
+        }
+        this.details.style.setProperty("transition", "height 0.35s");
```

Comment:
Can you elaborate on the goals here - seems costly to setup. What's the value being added by running this as part of the component API?

Reply:
It can be a part of the style file itself, however, I noticed that it would by default animate on the first render as well irrespective of the expanded or collapsed state.

Reply:
That sounds like something that we can work through without needing to call GBCR a few times, hiding, then showing the control. Animating height is difficult regardless. Can you articulate the end goal you are going for and we can see if we can help?

Reply:
The idea is to calculate the heights dynamically when the disclosure is expanded and collapsed. These heights are calculated once on `connectedCallback` and stored for later use. However, I was able to remove line no. 85 and add a transition in style file itself after removing the card from the demo.

Reply:
@chrisdholt, the changes are made. Could you review one more time? Do let me know in case more changes expected. 😄